### PR TITLE
Rails React

### PR DESCRIPTION
**Pull request by Jeddaa**
Dear Reviewer,

In this PR, I have added the required tasks to configure react with a webpack on a ruby on rails app.

### CODE REVIEW

File: app/javascript/app/greetingSlice.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+const initialState = {
+  greeting: {},
+  isLoading: false,
+  isError: false
+};
+
+const URL = 'http://localhost:3000/api/greetings';
+
```
Comment (KDlamini) [inline app/javascript/app/greetingSlice.js:12]: 🚀 Great job setting up your project. Everything is configured well in your project 💯 

However, there is an issue with displaying the greeting on the page. The API endpoint is set up correctly and returns a random greeting as expected. The issue seems to be with CORS preventing the HTTP request from redux.

image

Here's one way you can resolve this, although you are free to find another solution if this one doesn't work for you.

1. The easiest way to configure CORS on your Rails app is to use [rack-cors gem](https://github.com/cyu/rack-cors). You can install it by adding the following line to your Gemfile: 

```ruby
gem 'rack-cors'
```

2. Next, you need to provide the configuration for the gem. You need to inform Rails which origin it should allow. To do that, you need to create a new initializer for your application. Create the `cors.rb` file inside the `config/initializers` directory.

The content of the `config/initializers/cors.rb` should be similar to this: 

```ruby
Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins 'http://example.com:80' #modify the origin to be the URL of your API endpoint 
    resource '*', headers: :any, methods: [:get, :post]
  end
end
```

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @Jeddaa 🙋‍♀️,  

Great work so far! 💯 
You have done a great job setting up a hello rails app with React and Rails API. 🎊 🥳 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## HIGHLIGHTS 🌟 

✔️ Professional Pull Request
✔️ API endpoint is setup and working correctly
✔️ UI is implemented using React and Redux
✔️ Database migrations are setup correctly

## Required Changes ♻️
 
Although you cannot change your commit history. Please keep this in mind when writing commit messages in the future.
Please make sure to follow good practices for writing commit messages. Here are some common guidelines for writing commit messages:

- Start with a Capital Letter: Commit messages should start with a capital letter, just like any other sentence or title. This makes them more readable and consistent with general writing conventions.
- Use Present Tense: Write your commit messages in the present tense, as if you're describing what the commit does at the moment it's applied. For example, "Add feature" instead of "Added feature" or "Adding feature."
- Provide Context: Your commit message should briefly explain why the change is being made and any relevant context. This helps others understand the purpose of the commit without having to delve into the code changes.
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/app/greetingSlice.js (hunk 1 above)
Comment (Jeddaa) [reply app/javascript/app/greetingSlice.js:12]: thank you for catching that as i didn't have that issue

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Jeddaa  

Your project is complete! Great job for making the changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
✔️Descriptive Pull Request
✔️ API endpoint is setup and working correctly
✔️ UI is implemented using React and Redux
✔️ No linter errors


---[optional] Please like the previous reviewer noted, always ensure your Commit messages should start with a capital letter, just like any other sentence or title. This makes them more readable and consistent with general writing conventions.
<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**